pallets/validators-manager/src/benchmarking.rs

```diff
@@ -10,102 +10,129 @@ use super::*;
+
+const NEW_COLLATOR_ETHEREUM_PUBLIC_KEY: [u8; 33] =
+    hex!["03f171af36531200540b2badee5ed581b0a51f4e4a1a995025e149b9721b050074"];
+
+// Derived from [7u8;32] private key
+const VOTER_SENDER_ETHEREUM_COLLATOR_PUBLIC_KEY: [u8; 33] =
+    hex!["02989c0b76cb563971fdc9bef31ec06c3560f3249d6ee9e5d83c57625596e05f6f"];
+
+const MINIMUM_ADDITIONAL_BENCHMARKS_VALIDATORS: usize = 2;
+
+const MNEMONIC: &str =
+    "basic anxiety marine match castle rival moral whisper insane away avoid bike";
+
+fn generate_resigning_collator_account_details<T: Config>(
+) -> (T::AccountId, <T as pallet_avn::Config>::AuthorityId, Public) {
+    let key = <T as avn::Config>::AuthorityId::generate_pair(Some(MNEMONIC.as_bytes().to_vec()));
 
-    // Setup resigner account id and key
-    let resigner: Validator<T::AuthorityId, T::AccountId> = validators[1].clone();
     let mut resigner_account_bytes: [u8; 32] = [0u8; 32];
     resigner_account_bytes
-        .copy_from_slice(&hex!("1ed1aadead9704b693af012a9f24e1f00dc7e2a0b4eb99f9e0bc0c35a8d20223"));
-    let resigner_account_id =
-        T::AccountId::decode(&mut &resigner_account_bytes.encode()[..]).unwrap();
-    validators[1] = Validator::new(resigner_account_id, resigner.key);
+        .copy_from_slice(&hex!("0e08c08252ef2a5cbeb5d6a0dcaa1b8354aded57d603cdc4adc5e27709a6904f"));
+    (
+        T::AccountId::decode(&mut &resigner_account_bytes.encode()[..]).unwrap(),
+        key,
+        Public::from_raw(RESIGNING_COLLATOR_ETHEREUM_PUBLIC_KEY),
+    )
+}
 
-    // Setup validators in avn pallet
-    avn::Validators::<T>::put(validators.clone());
+fn generate_sender_collator_account_details<T: Config>(
+) -> (T::AccountId, <T as pallet_avn::Config>::AuthorityId, Public) {
+    let key = <T as avn::Config>::AuthorityId::generate_pair(Some(MNEMONIC.as_bytes().to_vec()));
 
-    validators
-        .iter()
-        .enumerate()
-        .for_each(|(i, v)| force_add_collator::<T>(&v.account_id, i as u64));
+    let mut resigner_account_bytes: [u8; 32] = [0u8; 32];
+    resigner_account_bytes
+        .copy_from_slice(&hex!("b41f90b123b66c18f0f869f3b9ae8a09d118419f8736240fcc7b8256517cc233"));
+    (
+        T::AccountId::decode(&mut &resigner_account_bytes.encode()[..]).unwrap(),
+        key,
+        Public::from_raw(VOTER_SENDER_ETHEREUM_COLLATOR_PUBLIC_KEY),
+    )
+}
+
+// Add additional collators, on top of genesis configuration
+fn setup_additional_validators<T: Config>(number_of_additional_validators: u32) {
+    assert!(number_of_additional_validators >= MINIMUM_ADDITIONAL_BENCHMARKS_VALIDATORS as u32);
+
+    let mut avn_validators: Vec<Validator<<T as pallet_avn::Config>::AuthorityId, T::AccountId>> =
+        Vec::new();
+
+    let mut validators: Vec<(T::AccountId, Public)> = Vec::new();
+    let vote_sender_index = number_of_additional_validators - (1 as u32);
+
+    for i in 0..number_of_additional_validators {
+        let (account, key, eth_key) = match i {
```

Comment:
Could we make clear whether `key` is a private or public key and on which chain it is?

Reply:
Updated